docs/install/pubsub-service-account.md

```diff
@@ -56,36 +56,10 @@ its improved security properties and manageability. For more information about
 Workload Identity see
 [here](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity).
 
-1. Enable Workload Identity. Skip this step if you have already enabled it
-   during the control plane setup:
-   [Install Knative-GCP](install-knative-gcp.md).
-
-    1. Ensure that you have enabled the Cloud IAM Service Account Credentials API.
-       ```shell
-       gcloud services enable iamcredentials.googleapis.com
-       ```
-    1. If you didn't enable Workload Identity when you
-    created your cluster, run the following commands: 
-    
-       Modify the cluster to enable Workload Identity first:
-    
-       ```shell 
-       gcloud container clusters update $CLUSTER_NAME \
-         --workload-pool=$PROJECT_ID.svc.id.goog
-       ```
-       Then, create a new node pool or modify an existing node pool to enable GKE_METADATA.
-       Here, we assume that you are using an existing node pool. 
-       You can get `POOL_NAME` by `gcloud container node-pools list --cluster=${CLUSTER_NAME}`, 
-       and choose to update a specific node pool or all node pools by:
-       
-       ```shell 
-       gcloud container node-pools update $POOL_NAME \
-          --cluster=$CLUSTER_NAME \
-          --workload-metadata=GKE_METADATA
-       ```
-       ***NOTE***: These commands may take a long time to finish. Check 
-       [Enable Workload Identity on an existing cluster](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity#enable_on_an_existing_cluster)
-       for more information.
+1. Enable Workload Identity. Checking [Configure Authentication Mechanism](authentication-mechanisms-gcp.md) ->
```

Comment:
```suggestion
1. [Enable Workload Identity](authentication-mechanisms-gcp.md/#option-1-recommended-workload-identity)
```

Reply:
This just points to option1, but user only need to configure first section of option1. I'd rather use more descriptions to make user less confused. like  
```
Checking [Configure Authentication Mechanism](authentication-mechanisms-gcp.md) ->
[Option 1 (Recommended): Workload Identity]([Enable Workload Identity](authentication-mechanisms-gcp.md/#option-1-recommended-workload-identity)) -> 
Enable Workload Identity, for setup details. 

```

Reply:
I'd rather the user use WI everywhere (for both the controllers and the data plane), at least in our guide. So if they decide to use WI, they should repeat all steps for the control plane.

If the users know what they are doing, they can do things like using secret in the controller while WI for data plane, but I highly doubt why they want that.

Reply:
Sorry I didn't make it clear when I suggested the change. I think we should try to make the docs easier to follow, instead of being too flexible. As a developer, I need to read the docs really carefully to understand what's going on, I am worried that our users will get even more confused.

So let's try to simplify the doc to have some opinionated way to install. Power users can figure out flexibilities if they really want to.

Reply:
Here, We only need section1 (for this data plane set up doc) of Option 1 (Recommended): Workload Identity, the left sections are for configuring control plane. If user have enabled WI in control plane, they will skip this step. Only users set secret in control plane and want to use WI in data plane will go to this step.

Reply:
> So let's try to simplify the doc to have some opinionated way to install. 

Could you elaborate more about this?  I don't think we should lead user to set WI for control plane in this doc (doc for setting data plane), this is like making user feel that they have to set up control plane of WI to make WI in data plane work. 
 
> Power users can figure out flexibilities if they really want to.

Actually, this section are just giving the power users instruction. Common user will use init script for control plane setup, and stick to one configuration method for both control plane and data plane (they will probably skip this step).

Reply:
> Only users set secret in control plane and want to use WI in data plane will go to this step.

That's my question. Is there a real use case that a user wants secret for control plane but WI for data plane? We don't have to answer this in the PR though.

> Actually, this section are just giving the power users instruction. Common user will use init script for control plane setup, and stick to one configuration method for both control plane and data plane (they will probably skip this step).

I am fine with this. Ideally I prefer sections for power users in a separate doc and is only referenced in the main doc so the main doc looks much cleaner and not intimidating to a first time user.

Reply:
@liu-cong I created issue for separating doc to make main doc simple #945